onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -320,36 +324,15 @@ Status PrepareForReduce(const Tensor* X,
   std::vector<bool> reduced(rank, false);
   prepare_reduce_metadata.output_dims.reserve(input_dims.size());
   if (axes.size() > 0) {
-    int64_t reduced_axis;
-    std::vector<uint64_t> reduced_axes(axes.size());
     prepare_reduce_metadata.output_dims = input_dims;
-    for (size_t i = 0; i < axes.size(); i++) {
-      reduced_axis = axes[i];
-      const int64_t axis = HandleNegativeAxis(reduced_axis, rank);
+    for (auto axis : axes) {
+      axis = HandleNegativeAxis(axis, rank);
       ORT_ENFORCE(input_dims[axis] != 0,
                   "Can't reduce on dim with value of 0 if 'keepdims' is false. "
                   "Invalid output shape would be produced. input_shape:",
                   input_shape);
       prepare_reduce_metadata.output_dims[axis] = 1;
       reduced[axis] = true;
-      reduced_axes[i] = axis;
-    }
-
-    bool contiguous_axes = true;
```

Comment:
Have we tested this for Bart? @ashbhandare - can you pls review this part? thx

Reply:
FYI, the contiguous axes detection moved here:
[https://github.com/microsoft/onnxruntime/blob/874e134401659b508bda653207322579405e34e3/onnxruntime/core/providers/cuda/reduction/reduction_functions.cc#L16](https://github.com/microsoft/onnxruntime/blob/874e134401659b508bda653207322579405e34e3/onnxruntime/core/providers/cuda/reduction/reduction_functions.cc#L16)

---
In reply to: [516860636](https://github.com/microsoft/onnxruntime/pull/5639#discussion_r516860636) [](ancestors = 516860636)